### ISSUE TITLE
Fenix asks for camera permissions just to make me upload files on some websites

### ISSUE THREAD

**Issue by Palloxin** [mozilla-mobile/fenix#27646]
### Steps to reproduce

https://user-images.githubusercontent.com/75091899/199284887-4691fc17-b9fc-4bfa-a8ac-e31dd2771442.mp4

### Expected behaviour

should not ask camera permissions, and just open the archive of my files since fenix has the storage permission

### Actual behaviour

fenix asks camera permission to upload file despite it having already storage permissions

### Device name

Huawei p8 lite 2017

### Android version

Android 8 Miui 8

### Firefox release type

Firefox Nightly

### Firefox version

106.1.0

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/FNXV2-21909)

**Comment by Mugurell** [comment]
Thank you for the report!
Can you please post the exact URL and complete steps for reproducing?
Does the issue happens on other browsers also?

**Comment by Palloxin** [comment]
-https://yandex.com/images then follow the steps in the video, the touch of the finger is visible.
-i don't know, this has been happening for a long time. i reported just because a was very tired of it. since i know about this issue for a long time i doubt this happens on other android browsers

**Comment by Mugurell** [comment]
Thank you for the details!
Seems like the website does allow searching by image by tapping on this button

![image](https://user-images.githubusercontent.com/11428869/199287164-9ca625c7-c14c-4d1d-b760-a674833e85e9.png)
 
 The image used for searching can be an existing one from the available storage or a new one taken from camera.
 
 Testing on Chrome I see that the camera permission is asked only when the user chooses the camera as the image source:
 
 

https://user-images.githubusercontent.com/11428869/199288780-e6ae9067-e687-4ff0-b460-ec2625e9e202.mp4

**Comment by Palloxin** [comment]
i see, but i would like fenix to allow me to upload the image/files without asking for camera permission

**Comment by Mugurell** [comment]
Agree that we should try to avoid asking for the camera permission beforehand.

**Comment by DreVla** [comment]
I have investigated this bug and traced the problem all the way to the FilePicker in feature-prompts package in Android Components. The issue seems to be that when asked to handle a file request, the picker iterates through all of the MimeTypes and checks if there are any permissions required in order to display multiple choices in the chooser. It should ask for these permissions only if the user selects certain option from chooser (camera for example). Here is one solution I have tried but with no avail:
https://stackoverflow.com/questions/44822840/request-permission-after-selected-optioneither-camera-or-gallery-from-intent-c

Some alternatives would be to implement a similar screen like chrome and some other browser do or use a different picker, similar to how duck duck go does it.

Chrome example (custom screen with possibility to take a picture with camera, browse more pictures or quickly select from most recent photos):

https://user-images.githubusercontent.com/32488956/200833508-40fe73fa-b6d9-4b33-b7ab-6a6af5a9802e.mp4


Duck Duck Go example, no permission required, going straight into android file picker, having recent images:

https://user-images.githubusercontent.com/32488956/200834363-c0a2da94-72da-4947-afc6-9899f694512a.mp4
